frame.go

```diff
@@ -1151,18 +1160,28 @@ func (f *framer) readByte() byte {
 }
 
 func (f *framer) readInt() (n int) {
+	if len(f.rbuf) < 4 {
+		panic(fmt.Errorf("not enough bytes in buffer to read int require 4 got: %d", len(f.rbuf)))
```

Comment:
The panics are caught and if they are not a `runtime.Error` then they are assigned as the error return from `parseFrame` so that they propagate out to the caller without having to have lots of control flow in the parse chain. It's the same mechanism that the standard library uses in the same situations such as in json encoding/decoding.